pkg/metrics/util.go

```diff
@@ -42,3 +49,51 @@ func GetBuildType() string {
 	}
 	return "internal"
 }
+
+func validatePort(setting env.Setting) error {
+	val := setting.Setting()
+	addr, err := net.ResolveTCPAddr("tcp", val)
+	if err != nil {
+		return err
+	}
+	log.Debugf("%s=%s, resolved to %+v", setting.EnvVar(), val, addr)
+	return nil
+}
+
+func validateTLS() error {
+	certFile := filepath.Join(env.SecureMetricsCertDir.Setting(), env.TLSCertFileName)
+	if ok, err := fileutils.Exists(certFile); !ok {
+		if err != nil {
+			log.Errorf("failed to validate file %q: %s", certFile, err.Error())
+		}
+		return errors.Wrapf(errox.NotFound, "secure metrics certificate file %q not found", certFile)
+	}
+
+	keyFile := filepath.Join(env.SecureMetricsCertDir.Setting(), env.TLSKeyFileName)
+	if ok, err := fileutils.Exists(keyFile); !ok {
+		if err != nil {
+			log.Errorf("failed to validate file %q: %s", keyFile, err.Error())
+		}
+		return errors.Wrapf(errox.NotFound, "secure metrics key file %q not found", keyFile)
+	}
+	return nil
+}
+
+// ValidateMetricsSetting returns an error if the environment variable is invalid.
+func ValidateMetricsSetting() error {
+	if !env.MetricsEnabled() {
+		return nil
+	}
+	return validatePort(env.MetricsPort)
+}
+
+// ValidateSecureMetricsSetting returns an error if the environment variable is invalid.
+func ValidateSecureMetricsSetting() error {
```

Comment:
nit: Could these functions be unexported if the only use is within the package ?